packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -155,261 +165,299 @@ function getFirstSlottedMatch(host: HTMLElement, nodeList: NodeList): Element |
-    // properly in jsdom, we need to make sure those methods doesn't respect the shadow DOM
-    // semantic.
-    // https://github.com/salesforce/lwc/pull/1179#issuecomment-484041707
-    return class PatchedHTMLElement extends Ctor {
-        // Regular Elements
-        querySelector(this: Element, selector: string): Element | null {
-            return lightDomQuerySelector(this, selector);
-        }
-        querySelectorAll(this: Element, selectors: string): NodeListOf<Element> {
-            return createStaticNodeList(lightDomQuerySelectorAll(this, selectors));
-        }
-        get innerHTML(this: Element): string {
-            const childNodes = getInternalChildNodes(this);
-            let innerHTML = '';
-            for (let i = 0, len = childNodes.length; i < len; i += 1) {
-                innerHTML += getOuterHTML(childNodes[i]);
-            }
-            return innerHTML;
-        }
-        set innerHTML(this: Element, value: string) {
-            innerHTMLSetter.call(this, value);
-        }
-        get outerHTML() {
-            return getOuterHTML(this);
-        }
-        set outerHTML(this: Element, value: string) {
-            outerHTMLSetter.call(this, value);
-        }
+function firstElementChildGetterPatched(this: ParentNode) {
+    return this.children[0] || null;
+}
 
-        // CE patches
-        attachShadow(options: ShadowRootInit): SyntheticShadowRootInterface {
-            return attachShadow(this, options) as SyntheticShadowRootInterface;
-        }
-        addEventListener(
-            this: EventTarget,
-            type: string,
-            listener: EventListener,
-            options?: boolean | AddEventListenerOptions
-        ) {
-            if (hasSyntheticShadow(this as HTMLElement)) {
-                addCustomElementEventListener(this as HTMLElement, type, listener, options);
-            } else {
-                superAddEventListener.call(this as HTMLElement, type, listener, options);
+function lastElementChildGetterPatched(this: ParentNode) {
+    const { children } = this;
+    return children.item(children.length - 1) || null;
+}
+
+// Non-deep-traversing patches
+defineProperties(Element.prototype, {
+    innerHTML: {
+        get(this: Element): string {
+            if (isNodeShadowed(this) || hasSyntheticShadow(this)) {
+                return innerHTMLGetterPatched.call(this);
             }
-        }
-        removeEventListener(
-            this: EventTarget,
-            type: string,
-            listener: EventListener,
-            options?: boolean | AddEventListenerOptions
-        ) {
-            if (hasSyntheticShadow(this as HTMLElement)) {
-                removeCustomElementEventListener(this as HTMLElement, type, listener, options);
-            } else {
-                superRemoveEventListener.call(this as HTMLElement, type, listener, options);
+            // TODO: make this a global patch with a way to disable it
```

Comment:
@ravijayaramappa @ekashida this is actually important... we have few more global patches that we don't have in place today... and I'm still questioning myself if our current escape hatch should apply to those apis (similar to what we do with querySelectors and others when executed outside of the scope of LWC), the apis in question are:

Note: cloneNode, contains, compareDocumentPosition, textContent
Element: outerHTML, innerHTML

I think we can probably add it to cloneNode, contains, compareDocumentPosition, but the other 3 seems like not a problem, unless you're doing something very wrong at the aura level... to get the content and test it or something, what do you think?

/cc @trevor-bliss in case he knows.

Reply:
I feel like we should have the escape hatch for these APIs. They are high traffic and in the likely hood of a failures, the escape hatch gives us some wiggle room.
```Node: cloneNode, textContent
Element: outerHTML, innerHTML```

Reply:
✅